src/crimson/osd/object_context_loader.cc

```diff
@@ -144,28 +144,27 @@ using crimson::common::local_conf;
                                        bool existed)
   {
     LOG_PREFIX(ObjectContextLoader::get_or_load_obc);
-    auto loaded =
-      load_obc_iertr::make_ready_future<ObjectContextRef>(obc);
-    if (existed) {
-      if (!obc->is_loaded_and_valid()) {
-	ERRORDPP(
-	  "obc for {} invalid -- fully_loaded={}, "
-	  "invalidated_by_interval_change={}",
-	  dpp, obc->get_oid(),
-	  obc->fully_loaded, obc->invalidated_by_interval_change
-	);
+    DEBUGDPP("{} -- fully_loaded={}, "
+             "invalidated_by_interval_change={}",
+             dpp, obc->get_oid(),
+             obc->fully_loaded,
+             obc->invalidated_by_interval_change);
+    return with_shared(obc->loading_mutex, [this, obc, existed, FNAME] {
```

Comment:
seastar::with_shared?

Reply:
@cyx1231st Maybe SeaStore's `Cache::get_extent()`-like methods can also follow this approach?

Reply:
@xxhdx1985126 Sorry I'm not following. Which mutex/lock are you refering to in the `Cache::get_extent()` related path?

Reply:
> @xxhdx1985126 Sorry I'm not following. Which mutex/lock are you refering to in the `Cache::get_extent()` related path?

Sorry, I didn't make myself clear. With regard to `SeaStore::Cache`, we have a similar issue and we need to make sure that the check for whether an extent is in `Cache` and the load of the missing extent are synchronous. Maybe we can take an approach like this pr, adding an mutex/lock for each extent and lock it when trying to load the extent from disk. What do you think?

Reply:
If there is no specific blocking problem that the current way cannot cover (or it result in unacceptabe complexity). Probably it's better to keep the existing way.

The mutex added in this case is because otherwise the code will break.